File: projects/c_programming/lvl_1/2.c
//WAP to input two numbers from the keyboard and print their average

#include <stdio.h>

int main()
{
    int a, b, avg;
    printf("Enter first number: ");
    scanf("%d", &a);
    printf("Enter second number: ");
    scanf("%d", &b);
    avg = (a + b) / 2;
    printf("Their average is %d\n", avg);
}

File: projects/rust/lvl_1/2.rs
// WAP a program to input two numbers from the keyboard and print their average

use std::io;

fn main() {
    // Declare the variables we'll need
    let (mut a, mut b, mut avg) = (String::new(), String::new(), 0);
    // Getting user input
    println!("Enter the first number: ");
    io::stdin().read_line(&mut a).expect("Error reading input");
    println!("Enter the second number: ");
    io::stdin().read_line(&mut b).expect("Error reading input");
    // Converting the strings to integers
    let num1: i32 = a.trim().parse().ok().expect("Error parsing input");
    let num2: i32 = b.trim().parse().ok().expect("Error parsing input");
    avg = (num1 + num2) / 2;

    println!("The average is {}", avg);
}